pkg/argocd/update_test.go

```diff
@@ -3643,3 +3643,58 @@ func Test_GetRepositoryLock(t *testing.T) {
 	require.NotNil(t, state.repositoryLocks[repo2])
 	require.Equal(t, lock3, state.repositoryLocks[repo2])
 }
+
+func Test_mergeKustomizeOverride(t *testing.T) {
+	tests := []struct {
+		name     string
+		existing v1alpha1.KustomizeImages
+		new      v1alpha1.KustomizeImages
+		expected v1alpha1.KustomizeImages
+	}{
+		{"with-tag", []v1alpha1.KustomizeImage{"nginx:foo"},
+			[]v1alpha1.KustomizeImage{"nginx:foo"},
+			[]v1alpha1.KustomizeImage{"nginx:foo"}},
+		{"with-tag-1", []v1alpha1.KustomizeImage{"nginx"},
+			[]v1alpha1.KustomizeImage{"nginx:latest"},
+			[]v1alpha1.KustomizeImage{"nginx:latest"}},
+		{"with-tag-sha", []v1alpha1.KustomizeImage{"nginx:latest"},
+			[]v1alpha1.KustomizeImage{"nginx:latest@sha256:91734281c0ebfc6f1aea979cffeed5079cfe786228a71cc6f1f46a228cde6e34"},
+			[]v1alpha1.KustomizeImage{"nginx:latest@sha256:91734281c0ebfc6f1aea979cffeed5079cfe786228a71cc6f1f46a228cde6e34"}},
+
+		{"2-images", []v1alpha1.KustomizeImage{"nginx:latest",
```

Comment:
Cheng, do you have an example where the new image tag replaces the old image tag to test out the conditional statements in your new code in update.go?

Reply:
the sub-tests: `with-tag-1`, `with-tag-sha`, and `2-images` all replace the existing tag with the new tag.

Reply:
I rechecked this again and yeah, the first test `with-tag`, the existing and the new tag are the same.  The next three tests, by definition of your test case struct, the tags are different.